macros/tests/ui/gprogram_works.rs

```diff
@@ -0,0 +1,39 @@
+use sails_macros::gprogram;
+
+struct MyProgram;
+
+#[gprogram]
+impl MyProgram {
+    pub fn new(config: u32) -> Self {
+        let _config = config;
+        Self
+    }
+
+    pub fn service1(&self) -> svc::MyService {
+        svc::MyService
+    }
+}
+
+pub mod svc {
+    #![allow(dead_code)]
```

Comment:
Why do we need this one?

Reply:
To silent the compiler, service methods are not used in test. Check only gprogram.